packages/rlc-common/src/buildObjectTypes.ts

```diff
@@ -417,18 +421,31 @@ export function getPropertySignature(
   importedModels: Set<string>
 ): PropertySignatureStructure {
   const propertyName = property.name;
-
   const description = property.description;
-  let type =
-    generateForOutput(schemaUsage, property.usage) && property.outputTypeName
-      ? property.outputTypeName
-      : property.typeName
-      ? property.typeName
-      : property.type;
-  if (property.typeName && property.fromCore) {
-    importedModels.add(property.typeName);
+  let type;
+  if (
+    property.type === "array" &&
+    (property as any).items &&
```

Comment:
It would be good to leverage the actual type for the calculation to avoid using 'as any'.
I think this is another area we shall improve our codebase.

Reply:
I created an issue as our MQ item https://github.com/Azure/autorest.typescript/issues/1993